packages/api-nest/src/repositories/reading-progress.repository.ts

```
@@ -0,0 +1,140 @@
+ * Handles all data access operations for sentinel-based reading progress
+ */
+@Injectable()
+export class ReadingProgressRepository implements IReadingProgressRepository {
+  constructor(
+    @InjectRepository(ReadingProgressEntity)
+    private readonly repository: Repository<ReadingProgressEntity>,
+  ) {}
+
+  /**
+   * Find reading progress for a specific user + item + content version
+   */
```

ESLint

[error] Missing JSDoc @returns for function.

(valid-jsdoc)